base/socket.jl

```diff
@@ -577,8 +587,16 @@ end
 
 function getaddrinfo(cb::Function, host::ASCIIString)
     callback_dict[cb] = cb
-    uv_error("getaddrinfo",ccall(:jl_getaddrinfo, Int32, (Ptr{Void}, Cstring, Ptr{UInt8}, Any, Ptr{Void}),
-                                 eventloop(), host, C_NULL, cb, uv_jl_getaddrinfocb::Ptr{Void}))
+    status = ccall(:jl_getaddrinfo, Int32, (Ptr{Void}, Cstring, Ptr{UInt8}, Any, Ptr{Void}),
+                   eventloop(), host, C_NULL, cb, uv_jl_getaddrinfocb::Ptr{Void})
+    if status == UV_EINVAL
+        throw(ArgumentError("Invalid uv_getaddrinfo() agument"))
+    elseif status in [UV_ENOMEM, UV_ENOBUFS]
```

Comment:
It is [in this case](https://github.com/libuv/libuv/blob/v1.x/src/win/getaddrinfo.c#L305).